pytest_pyvista/pytest_pyvista.py

```diff
@@ -197,14 +198,15 @@ def __call__(self, plotter):  # noqa: ANN001, ANN204, C901, PLR0912
         image_name = test_name[5:] + ".png"
         image_filename = os.path.join(self.cache_dir, image_name)  # noqa: PTH118
 
-        if not os.path.isfile(image_filename) and self.fail_extra_image_cache and not self.reset_image_cache:  # noqa: PTH113
-            # Make sure this doesn't get called again if this plotter doesn't close properly
-            plotter._before_close_callback = None  # noqa: SLF001
-            msg = f"{image_filename} does not exist in image cache"
-            raise RegressionFileNotFound(msg)
-
-        if ((self.add_missing_images and not os.path.isfile(image_filename)) or self.reset_image_cache) and not self.reset_only_failed:  # noqa: PTH113
-            plotter.screenshot(image_filename)
+        overwrite_cache = (self.add_missing_images or self.reset_image_cache) and not self.reset_only_failed
+        if not Path(image_filename).is_file() or overwrite_cache:
+            if self.allow_unused_generated or overwrite_cache:
```

Comment:
I personally find the kludging of these two type of uses together in a set of nested if blocks to be confusing to reason with in my head.  Maybe related to below as well.

This logic means that if `self.allow_unused_generated` is `True` and there is no image cache file,  then it adds the image cache file?  Shouldn't it be required to also be using `self.add_missing_images = True` or `self.reset_only_failed` in this case?

Reply:
What is optimal in understanding is definitely subjective.  I find the separate usage more understandable in that the first blocks checks if there should be an error raised.  The second block only checks if an image should be generated.  The logic then follows from those goals.  When combined, there are competing objectives in each branch for which outcome you are trying to understand.